qiskit_machine_learning/neural_networks/effective_dimension.py

```diff
@@ -0,0 +1,347 @@
+                ``(num_weight_samples, num_weights)``, or an ``int`` to indicate the number of
+                parameter sets to sample randomly from a uniform distribution. By default,
+                ``weight_samples = 1``.
+            input_samples: An array of samples to the neural network, of shape
+                ``(num_input_samples, qnn_input_size)``, or an ``int`` to indicate the number of
+                input sets to sample randomly from a normal distribution. By default,
+                ``input_samples = 1``.
+            callback: A callback function for the Monte Carlo sampling.
+        """
+
+        # Store arguments
+        self._weight_samples = None
+        self._input_samples = None
+        self._num_weight_samples = 1
+        self._num_input_samples = 1
+        self._model = qnn
+        self._callback = callback
+
+        # Define weight samples and input samples
+        self.weight_samples = weight_samples  # type: ignore
+        # input setter uses self._model
+        self.input_samples = input_samples  # type: ignore
+
+    @property
+    def weight_samples(self) -> np.ndarray:
+        """Returns network weight samples."""
+        return self._weight_samples
+
+    @weight_samples.setter
+    def weight_samples(self, weight_samples: Union[List[float], np.ndarray, int]) -> None:
+        """Sets network weight samples."""
+        if isinstance(weight_samples, int):
+            # random sampling from uniform distribution
+            self._weight_samples = algorithm_globals.random.uniform(
+                0, 1, size=(weight_samples, self._model.num_weights)
+            )
+        else:
+            weight_samples = np.asarray(weight_samples)
+            if len(weight_samples.shape) > 2 or weight_samples.shape[1] != self._model.num_weights:
+                raise QiskitMachineLearningError(
+                    f"The Effective Dimension class expects"
+                    f" a weight_samples array of shape (M, qnn.num_weights)."
+                    f" Got {weight_samples.shape}."
+                )
+            self._weight_samples = weight_samples
+
+        self._num_weight_samples = len(self._weight_samples)
+
+    @property
+    def input_samples(self) -> np.ndarray:
+        """Returns network input samples."""
+        return self._input_samples
+
+    @input_samples.setter
+    def input_samples(self, input_samples: Union[List[float], np.ndarray, int]) -> None:
+        """Sets network input samples."""
+        if isinstance(input_samples, int):
+            # random sampling from normal distribution
+            self._input_samples = algorithm_globals.random.normal(
+                0, 1, size=(input_samples, self._model.num_inputs)
+            )
+        else:
+            input_samples = np.asarray(input_samples)
+            if len(input_samples.shape) > 2 or input_samples.shape[1] != self._model.num_inputs:
```

Comment:
```suggestion
            if len(input_samples.shape) != 2 or input_samples.shape[1] != self._model.num_inputs:
```
Same as the weight_samples comment above

Reply:
OK, I see the point. Then, it's easier to restrict to 2D (other shapes are pretty rare and can be converted to 2D anyway) rather than changing the algorithm with unknown benefits.